File: BubblesDivePlanner/ViewModels/Plan/DiveModels/DiveProfile.cs
using BubblesDivePlanner.ViewModels.Model.Plan.DiveModels;

namespace BubblesDivePlanner.ViewModels.Plan.DiveModels
{
    public class DiveProfile : IDiveProfile
    {
        private readonly int compartments;

        public DiveProfile(int compartments)
        {
            this.compartments = compartments;
            MaxSurfacePressures = new double[compartments];
            NitrogenTissuePressures = new double[compartments];
            HeliumTissuePressures = new double[compartments];
            TotalTissuePressures = new double[compartments];
            ToleratedAmbientPressures = new double[compartments];
            AValues = new double[compartments];
            BValues = new double[compartments];
            CompartmentLoads = new double[compartments];
        }

        public double[] MaxSurfacePressures { get; }

        public double[] NitrogenTissuePressures { get; }

        public double[] HeliumTissuePressures { get; }

        public double[] TotalTissuePressures { get; }

        public double[] ToleratedAmbientPressures { get; }

        public double[] AValues { get; }

        public double[] BValues { get; }

        public double[] CompartmentLoads { get; }

        public double OxygenAtPressure { get; }

        public double NitrogenAtPressure { get; }

        public double HeliumAtPressure { get; }

        
    }
}

File: src/models/result/dive_profile.rs
use serde::{Deserialize, Serialize};

use crate::models::plan::dive_stage::DiveStage;

#[derive(PartialEq, Debug, Default, Clone, Serialize, Deserialize)]
pub struct DiveProfile {
    pub number_of_compartments: usize,
    pub maximum_surface_pressures: [f32; 16],
    pub compartment_loads: [f32; 16],
    pub nitrogen_tissue_pressures: [f32; 16],
    pub helium_tissue_pressures: [f32; 16],
    pub total_tissue_pressures: [f32; 16],
    pub tolerated_ambient_pressures: [f32; 16],
    pub a_values: [f32; 16],
    pub b_values: [f32; 16],
    pub oxygen_at_pressure: f32,
    pub helium_at_pressure: f32,
    pub nitrogen_at_pressure: f32,
    pub dive_ceiling: f32,
}

impl DiveProfile {
    pub fn new(number_of_compartments: usize) -> Self {
        Self {
            number_of_compartments,
            nitrogen_tissue_pressures: [
                0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79,
                0.79, 0.79,
            ],
            total_tissue_pressures: [
                0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79, 0.79,
                0.79, 0.79,
            ],
            ..Default::default()
        }
    }
}

#[cfg(test)]
mod dive_profile_should {}
